navigation/src/main/java/com/graphhopper/navigation/NavigateResponseConverter.java

```diff
@@ -319,11 +334,32 @@ private static void putBannerInstructions(InstructionList instructions, double d
         putSingleBannerInstruction(instructions.get(index + 1), locale, translationMap, primary);
 
         bannerInstruction.putNull("secondary");
-
-        if (instructions.size() > index + 2 && instructions.get(index + 2).getSign() != Instruction.REACHED_VIA) {
-            // Sub shows the instruction after the current one
-            ObjectNode sub = bannerInstruction.putObject("sub");
-            putSingleBannerInstruction(instructions.get(index + 2), locale, translationMap, sub);
+        if (instructions.get(index + 1).getInstructionDetails().isEmpty()) {
+            if (instructions.size() > index + 2 && instructions.get(index + 2).getSign() != Instruction.REACHED_VIA) {
+                // Sub shows the instruction after the current one
+                ObjectNode sub = bannerInstruction.putObject("sub");
+                putSingleBannerInstruction(instructions.get(index + 2), locale, translationMap, sub);
+            }
+        } else {
+            for (InstructionDetails details: instructions.get(index + 1).getInstructionDetails()) {
+                ObjectNode subBannerInstruction = bannerInstructions.addObject();
+                subBannerInstruction.put("distanceAlongGeometry", details.getBeforeTurn());
+        
+                ObjectNode subPrimary = subBannerInstruction.putObject("primary");
+                putSingleBannerInstruction(instructions.get(index + 1), locale, translationMap, subPrimary);
+
+                ObjectNode sub = subBannerInstruction.putObject("sub");
+                sub.put("text", "");
+                ArrayNode components = sub.putArray("components");
+                for (LaneInfo lane : details.getLanes()) {
+                    ObjectNode component = components.addObject();
+                    component.put("text", "");
+                    component.put("type", "lane");
+                    component.put("active", lane.isValid());
+                    ArrayNode directions = component.putArray("directions");
+                    putDirections(instructions.get(index + 1), lane, directions, primary, "active_direction");
```

Comment:
This requires some explanation, I checked the mapbox navigation API and how it works is:
when there's lane information to show it creates a bannerInstruction with the primary turn only, then it creates a second bannerInstruction with the same primary and a sub that contains the lane information.

Reply:
not yet, but I'll certainly do. I'll set up the environment and try it out.